File: TUnit.Templates.Tests/Snapshots/InstantiationTest.TUnit.AspireStarter._.verified/TUnit.AspireStarter/TUnit.AspireStarter.TestProject/GlobalSetup.cs
// Here you could define global logic that would affect all tests

// You can use attributes at the assembly level to apply to all tests in the assembly

using Aspire.Hosting;

[assembly: Retry(3)]
[assembly: System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]

namespace TUnit.AspireStarter.TestProject;

public class GlobalHooks
{
    public static DistributedApplication? App;
    public static ResourceNotificationService? ResourceNotificationService;

    [Before(TestSession)]
    public static async Task SetUp()
    {
        // Arrange
        var appHost = await DistributedApplicationTestingBuilder.CreateAsync<Projects.TUnit.AspireStarter_AppHost>();
        appHost.Services.ConfigureHttpClientDefaults(clientBuilder =>
        {
            clientBuilder.AddStandardResilienceHandler();
        });

        App = await appHost.BuildAsync();
        ResourceNotificationService = App.Services.GetRequiredService<ResourceNotificationService>();
        await App.StartAsync();
    }

    [After(TestSession)]
    public static void CleanUp()
    {
        Console.WriteLine("...and after!");
    }
}

File: TUnit.Templates/content/TUnit.VB/GlobalSetup.vb
' Here you could define global logic that would affect all tests

' You can use attributes at the assembly level to apply to all tests in the assembly
Imports TUnit.Core

<Assembly: Retry(3)>
<Assembly: CodeAnalysis.ExcludeFromCodeCoverage>

Public Class GlobalHooks

    <Before(HookType.TestSession)>
    Public Shared Sub SetUp()
        Console.WriteLine("Or you can define methods that do stuff before...")
    End Sub

    <After(HookType.TestSession)>
    Public Shared Sub CleanUp()
        Console.WriteLine("...and after!")
    End Sub

End Class